File: 002-n8n/scripts/import-workflows.py
#!/usr/bin/env python3
import sqlite3
import json
import os
import uuid
from datetime import datetime

DB_PATH = '/workspace/n8n/.n8n/database.sqlite'
WORKFLOWS_DIR = '/workspace/002-n8n/workflows'

# Connect to database
conn = sqlite3.connect(DB_PATH)
cursor = conn.cursor()

# Get all workflow JSON files
workflow_files = [f for f in os.listdir(WORKFLOWS_DIR) if f.endswith('.json')]
print(f"Found {len(workflow_files)} workflow files to import")

imported = 0
failed = 0

for file in workflow_files:
    try:
        file_path = os.path.join(WORKFLOWS_DIR, file)
        with open(file_path, 'r') as f:
            workflow_data = json.load(f)

        # Prepare workflow data
        workflow_id = str(uuid.uuid4())
        name = workflow_data.get('name', file.replace('.json', ''))
        active = 1 if workflow_data.get('active', True) else 0
        nodes = json.dumps(workflow_data.get('nodes', []))
        connections = json.dumps(workflow_data.get('connections', {}))
        settings = json.dumps(workflow_data.get('settings', {}))
        static_data = json.dumps(workflow_data.get('staticData')) if workflow_data.get('staticData') else None
        version_id = workflow_data.get('versionId', str(uuid.uuid4()))
        trigger_count = workflow_data.get('triggerCount', 0)
        created_at = datetime.now().strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]
        updated_at = created_at

        # Insert workflow
        cursor.execute('''
            INSERT INTO workflow_entity (
                id, name, active, nodes, connections, settings, staticData,
                versionId, triggerCount, createdAt, updatedAt, isArchived
            ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 0)
        ''', (
            workflow_id, name, active, nodes, connections, settings, static_data,
            version_id, trigger_count, created_at, updated_at
        ))

        print(f"✓ Imported {name} ({file}) - Active: {'Yes' if active else 'No'}")
        imported += 1

    except Exception as e:
        print(f"✗ Failed to import {file}: {str(e)}")
        failed += 1

# Commit and close
conn.commit()
conn.close()

print(f"\n{imported} workflows imported successfully, {failed} failed")

File: 002-n8n/scripts/import-workflows.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const sqlite3 = require('sqlite3').verbose();

const DB_PATH = '/workspace/n8n/.n8n/database.sqlite';
const WORKFLOWS_DIR = '/workspace/002-n8n/workflows';

// Open database
const db = new sqlite3.Database(DB_PATH, (err) => {
  if (err) {
    console.error('Error opening database:', err.message);
    process.exit(1);
  }
  console.log('Connected to n8n database');
});

// Get all workflow JSON files
const workflowFiles = fs.readdirSync(WORKFLOWS_DIR).filter(f => f.endsWith('.json'));

console.log(`Found ${workflowFiles.length} workflow files to import`);

let imported = 0;
let failed = 0;

// Import each workflow
workflowFiles.forEach((file, index) => {
  const filePath = path.join(WORKFLOWS_DIR, file);
  const workflowData = JSON.parse(fs.readFileSync(filePath, 'utf8'));

  const id = index + 1;
  const name = workflowData.name;
  const active = workflowData.active !== undefined ? (workflowData.active ? 1 : 0) : 1;
  const nodes = JSON.stringify(workflowData.nodes || []);
  const connections = JSON.stringify(workflowData.connections || {});
  const settings = JSON.stringify(workflowData.settings || {});
  const staticData = workflowData.staticData ? JSON.stringify(workflowData.staticData) : null;
  const tags = JSON.stringify(workflowData.tags || []);
  const createdAt = new Date().toISOString();
  const updatedAt = new Date().toISOString();

  const sql = `INSERT INTO workflow_entity (
    id, name, active, nodes, connections, settings, staticData, tags, createdAt, updatedAt
  ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`;

  db.run(sql, [id, name, active, nodes, connections, settings, staticData, tags, createdAt, updatedAt], (err) => {
    if (err) {
      console.error(`Failed to import ${file}:`, err.message);
      failed++;
    } else {
      console.log(`✓ Imported ${name} (${file}) - Active: ${active === 1 ? 'Yes' : 'No'}`);
      imported++;
    }

    // Close database after last workflow
    if (index === workflowFiles.length - 1) {
      setTimeout(() => {
        db.close((err) => {
          if (err) {
            console.error('Error closing database:', err.message);
          } else {
            console.log(`\n${imported} workflows imported successfully, ${failed} failed`);
          }
        });
      }, 100);
    }
  });
});
